pybalboa/balboa.py

```diff
@@ -237,379 +241,278 @@ async def send_panel_req(self, ba, bb):
+        7e1a0abf24 64dc 1400 4250323030304731 04 51800c6b 01 0a 0200 f9 7e <-- mine
+        7e1a0abf24 64c9 1300 4d51425035303120 01 0403daed 01 06 0400 35 7e
+        7e1a0abf24 64e1 2400 4d53343045202020 01 c3479636 03 0a 4400 19 7e
+        7e1a0abf24 64e1 1400 4250323130304731 11 ebce9fd8 03 0a 1600 d7 7e
+        7e1a0abf24 64dc 2400 4246425032305320 03 5cd4ccd7 01 0a 0000 de 7e
 
         """
 
-        model = [
-            data[9], data[10],
-            data[11], data[12], data[13],
-            data[14], data[15], data[16],
-        ]
-        model_name = "".join(map(chr, model))
-        self.model_name = model_name.strip()
-
-        self.cfg_sig = f"{data[18]:x}{data[19]:x}{data[20]:x}{data[21]:x}"
-        self.sw_vers = f"{str(data[7])}.{str(data[8])}"
+        self.sw_vers = f'{data[7]}.{data[8]}'
+        self.ssid = f'M{data[5]}_{data[6]} V{self.sw_vers}'
+        self.model_name = "".join(map(chr, data[9:17])).strip()
         self.setup = data[17]
-        self.ssid = f"M{str(data[5])}_{str(data[6])} V{self.sw_vers}"
-
-    def parse_config_resp(self, data):
-        """ Parse a config response.
-
-        SZ 02 03 04   05 06 07 08 09 10 11 12 13 14 15 16 17 18 19 20 21 22
-        1E 0A BF 94   02 14 80 00 15 27 37 EF ED 00 00 00 00 00 00 00 00 00
-        23 24 25 26 27 28 29 CB
-        15 27 FF FF 37 EF ED 42
-
-        22,23,24 seems to be mac prefix,  27-29 suffix.  25/26 unk
-        8-13 also full macaddr.
-        05 - nrof pumps. Bitmask 2 bits per pump.
-        06 - P6xxxxP5
-        07 - L1xxxxL2
-
-        I feel that the nrof pumps is untrustworthy here.
+        self.cfg_sig = f"{data[18]:x}{data[19]:x}{data[20]:x}{data[21]:x}"
+        self.voltage = self.VOLTAGE_240 if data[22] == 0x01 else self.VOLTAGE_UNKNOWN
+        self.heater_type = self.HEATERTYPE_STANDARD if data[23] == 0x0A else self.HEATERTYPE_UNKNOWN
+        self.dip_switch = f'{data[24]:08b}{data[25]:08b}'
+
+    def parse_setup_parameters(self, data):
+        """ Parse a setup parameters response.
+
+        01 02 03 04 05 06 07 08 09 10 11 12 13 14
+        ML AD PF PT 05 06 LL LH HL HH 11 12 13 CB
+        Bullfrog Stil7 / BWGWIFI1:
+        0E 0A BF 25 04 03 32 63 50 68 E9 01 45 4F
+
+        05-06 - unknown
+        07 = LL (low low) low range temperature's minimum
+        08 - LH (low high) low range temperature's maximum
+        09 - HL (high low) high range temperature's minimum
+        10 - HH (high high) high range temperature's maximum
+        11 - unknown
+        12 - number of pumps
+        13 - unknown
         """
 
-        macaddr = f'{data[8]:x}:{data[9]:x}:{data[10]:x}'\
-            f':{data[11]:x}:{data[12]:x}:{data[13]:x}'
+        self.tmin[0] = [data[7], self.to_celsius(data[7])]
+        self.tmax[0] = [data[8], self.to_celsius(data[8])]
+        self.tmin[1] = [data[9], self.to_celsius(data[9])]
+        self.tmax[1] = [data[10], self.to_celsius(data[10])]
```

Comment:
I'm slightly concerned about this.. we pull in tmax/tmin to celcius, but what if the spa is set in F?

Reply:
This stores both the Fahrenheit value, as well as the Celsius value of min and max temp values the spa supports. The spa, regardless of whether it is set to Fahrenheit or Celsius, always reports the value in Fahrenheit and then has to be converted to Celsius.

Reply:
I added a comment around that section to clarify